src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -737,8 +769,7 @@ private void PrepareForEntitiesDeletion(SaveChangesData result, IDictionary<stri
         {
             foreach (var deletedEntity in DeletedEntities)
             {
-                DocumentInfo documentInfo;
-                if (!DocumentsByEntity.TryGetValue(deletedEntity, out documentInfo)) continue;
+                if (!DocumentsByEntity.TryGetValue(deletedEntity, out DocumentInfo documentInfo)) continue;
```

Comment:
Done. Note that we didn't also validated that we have a defered put command but now we also checking that.